R/z_pages.R

```diff
@@ -23,34 +24,56 @@
 #' @examples
 #' \dontrun{
 #' library(animint2)
-#' p1 <- ggplot(mtcars, aes(x = mpg, y = wt)) +
-#'   geom_point()
-#' p2 <- ggplot(mtcars, aes(x = hp, y = wt)) +
-#'   geom_point()
-#' viz <- list(plot1 = p1, plot2 = p2)
-#' animint2pages(
-#'   viz,
-#'   github_repo = "my_animint2_plots",
-#'   commit_message = "New animint",
-#'   private = TRUE)
+#' mtcars$Cyl <- factor(mtcars$cyl)
+#' viz <- animint(
+#'   ggplot(mtcars, aes(x = mpg, y = disp, color=Cyl)) +
+#'     geom_point(),
+#'   ggplot(mtcars, aes(x = hp, y = wt, color=Cyl)) +
+#'     geom_point(),
+#'   title="Motor Trend Cars data viz",
+#'   source="https://github.com/animint/animint2/blob/master/R/z_pages.R"
+#' )
+#' animint2pages(viz, "animint2pages-example-mtcars")
 #' }
-#'
+#' 
 #' @export
-animint2pages <- function(plot.list, github_repo, owner=NULL, commit_message = "Commit from animint2pages", private = FALSE, required_opts = c("title","source"), ...) {
+animint2pages <- function(plot.list, github_repo, owner=NULL, commit_message = "Commit from animint2pages", private = FALSE, required_opts = c("title","source"), server=NULL, ...) {
   for(opt in required_opts){
     if(!opt %in% names(plot.list)){
       stop(sprintf("plot.list does not contain option named %s, which is required by animint2pages", opt))
     }
   }
-  # Check for required packages
+  ## Check for required packages
   for(pkg in c("gert", "gh")){
     if (!requireNamespace(pkg)) {
       stop(sprintf("Please run `install.packages('%s')` before using this function", pkg))
     }
   }
-  # Generate plot files
-  res <- animint2dir(plot.list, open.browser = FALSE, ...)
-  # Select non-ignored files to post
+  chrome.session <- chromote::ChromoteSession$new()
+  if (is.null(server)) {
+    res <- animint2dir(plot.list, open.browser = FALSE, ...)
+    portNum <- servr::random_port()
+    normDir <- normalizePath(res$out.dir, winslash = "/", mustWork = TRUE)
+    code = sprintf("servr::httd(dir='%s', port=%d)", normDir, portNum)
```

Comment:
this functionality (starting a server using servr::httd) seems duplicated here, and when printing animint, and in tests_init. is there some way to combine the repeated logic into a helper function?